plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -129,6 +145,18 @@
     AgentManager _agentMgr;
     @Inject
     NetworkOfferingDao _ntwkOfferingDao;
+    @Inject
+    ConfigurationDao _configDao;
+    @Inject
+    NuageVspManager _nuageVspManager;
+    @Inject
+    FirewallRulesDao _firewallRulesDao;
+    @Inject
+    FirewallRulesCidrsDao _firewallRulesCidrsDao;
+    @Inject
+    PhysicalNetworkDao _physicalNetworkDao;
+    @Inject
+    NetworkACLItemDao _networkACLItemDao;
 
     @Override
     public boolean applyIps(Network network, List<? extends PublicIpAddress> ipAddress, Set<Service> service) throws ResourceUnavailableException {
```

Comment:
This method is incredibly long and performing many different operations.  Decompose into a smaller private/utility methods and possibly consider some small ValueObjects.  Each of these methods should have a set of unit tests to verify them.